src/mono/wasm/runtime/http.ts

```diff
@@ -65,79 +65,36 @@ export function http_wasm_abort_response(res: ResponseExtension): void {
-            const copy = view.slice() as Uint8Array;
-            controller.enqueue(copy);
-        }
-        pull_promise_control.resolve();
-    }
-    catch (err) {
-        pull_state.dispose();
-        pull_promise_control.reject(err);
-    }
-    finally {
-        pull_state.__pull_promise_control = null;
-    }
+export function http_wasm_create_transform_stream(): TransformStreamExtension {
+    const transform_stream = new TransformStream<Uint8Array, Uint8Array>() as TransformStreamExtension;
+    transform_stream.__writer = transform_stream.writable.getWriter();
+    return transform_stream;
 }
 
-export function http_wasm_readable_stream_controller_error(pull_state: PullStateExtension, error: Error): void {
-    const controller = pull_state.__controller;
-    mono_assert(controller, "expected controller");
-    pull_state.__pull_promise_control?.reject(error);
-    pull_state.__fetch_promise_control?.reject(error);
-    pull_state.dispose();
-    controller.error(error);
-    pull_state.__pull_promise_control = null;
+export async function http_wasm_transform_stream_write(ts: TransformStreamExtension, bufferPtr: VoidPtr, bufferLength: number): Promise<void> {
+    mono_assert(bufferLength > 0, "expected bufferLength > 0");
+    mono_assert(ts.__fetch_promise_control, "expected fetch promise control");
+    await Promise.race([ts.__writer.ready, ts.__fetch_promise_control.promise]);
+    // the bufferPtr is pinned by the caller
+    const view = new Span(bufferPtr, bufferLength, MemoryViewType.Byte);
+    const copy = view.slice() as Uint8Array;
+    await Promise.race([ts.__writer.write(copy), ts.__fetch_promise_control.promise]);
 }
 
-export function http_wasm_fetch_stream(url: string, header_names: string[], header_values: string[], option_names: string[], option_values: any[], abort_controller: AbortController,
-    pull_delegate: (pull_state: PullStateExtension) => void,
-    pull_state: PullStateExtension): Promise<ResponseExtension> {
-    function pull(controller: ReadableByteStreamController): Promise<void> {
-        const { promise, promise_control } = createPromiseController<void>();
-        try {
-            mono_assert(!pull_state.__pull_promise_control, "expected pull_promise_control to be null");
-            pull_state.__controller = controller;
-            pull_state.__pull_promise_control = promise_control;
-            pull_delegate(pull_state);
-            return promise;
-        }
-        catch (error) {
-            pull_state.dispose();
-            pull_state.__controller = null;
-            pull_state.__pull_promise_control = null;
-            pull_state.__fetch_promise_control?.reject(error);
-            return Promise.reject(error);
-        }
-    }
-
-    function cancel(error: any) {
-        pull_state.__fetch_promise_control?.reject(error);
+export async function http_wasm_transform_stream_close(ts: TransformStreamExtension, error: Error | undefined): Promise<void> {
```

Comment:
This could be split into two methods, but it currently saves one export/import.

Reply:
Let's split it, for clarity sake